File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout/types/statistic/workouts_customer_statistic_response_dto.js
import {
    WorkoutAnswersStatisticResponseDTO
} from "./workout_answers_statistic_response_dto.js";

import {
    CustomerResponseDTO
} from "../../../customer/response/customer_response_dto.js";

export class WorkoutsCustomerStatisticResponseDTO {
    #numberOfMilliseconds;
    #numberOfWorkouts;
    #numberOfRounds;
    #customer;
    #workoutAnswersStatistic;

    constructor(json) {
        if (json) {
            this.#numberOfMilliseconds = json["number_of_milliseconds"];
            this.#numberOfWorkouts = json["number_of_workouts"];
            this.#numberOfRounds = json["number_of_rounds"];

            let customer = json["customer"];
            if (customer) {
                this.#customer = new CustomerResponseDTO(customer);
            }

            let workoutAnswersStatistic = json["workout_answers_statistic"];
            if (workoutAnswersStatistic) {
                this.#workoutAnswersStatistic = new WorkoutAnswersStatisticResponseDTO(workoutAnswersStatistic);
            }
        }
    }

    getNumberOfMilliseconds() {
        return this.#numberOfMilliseconds;
    }

    getNumberOfWorkouts() {
        return this.#numberOfWorkouts;
    }

    getNumberOfRounds() {
        return this.#numberOfRounds;
    }

    getCustomer() {
        return this.#customer;
    }

    getWorkoutAnswersStatistic() {
        return this.#workoutAnswersStatistic;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workout/types/statistic/customer/WorkoutsCustomerStatisticResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workout.types.statistic.customer;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.customer.response.CustomerResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.workout.types.statistic.WorkoutAnswersStatisticResponseDTO;

public class WorkoutsCustomerStatisticResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_milliseconds")
    private long numberOfMilliseconds;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_workouts")
    private long numberOfWorkouts;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_rounds")
    private long numberOfRounds;

    @JsonProperty("customer")
    private CustomerResponseDTO customer;

    @JsonProperty("workout_answers_statistic")
    private WorkoutAnswersStatisticResponseDTO workoutAnswersStatistic;

    public long getNumberOfMilliseconds() {
        return numberOfMilliseconds;
    }

    public void setNumberOfMilliseconds(long numberOfMilliseconds) {
        this.numberOfMilliseconds = numberOfMilliseconds;
    }

    public long getNumberOfWorkouts() {
        return numberOfWorkouts;
    }

    public void setNumberOfWorkouts(long numberOfWorkouts) {
        this.numberOfWorkouts = numberOfWorkouts;
    }

    public long getNumberOfRounds() {
        return numberOfRounds;
    }

    public void setNumberOfRounds(long numberOfRounds) {
        this.numberOfRounds = numberOfRounds;
    }

    public CustomerResponseDTO getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerResponseDTO customer) {
        this.customer = customer;
    }

    public WorkoutAnswersStatisticResponseDTO getWorkoutAnswersStatistic() {
        return workoutAnswersStatistic;
    }

    public void setWorkoutAnswersStatistic(WorkoutAnswersStatisticResponseDTO workoutAnswersStatistic) {
        this.workoutAnswersStatistic = workoutAnswersStatistic;
    }
}
